### PR TITLE
use small lock in following files

### PR DESCRIPTION
**Pull request by hujun260**
## Summary
use small lock in following files

arch/xtensa/src/esp32s3/esp32s3_rtc_lowerhalf.c
arch/z16/src/z16f/z16f_serial.c
arch/z80/src/z8/z8_serial.c
boards/arm/max326xx/max32660-evsys/src/max326_button.c boards/arm/stm32/stm32f4discovery/src/stm32_gs2200m.c drivers/i2c/i2c_bitbang.c
drivers/power/supply/act8945a.c
drivers/serial/uart_pci_16550.c
drivers/serial/uart_pl011.c
drivers/wireless/bluetooth/bt_uart_shim.c

## Impact
arch/xtensa/src/esp32s3/esp32s3_rtc_lowerhalf.c
arch/z16/src/z16f/z16f_serial.c
arch/z80/src/z8/z8_serial.c
boards/arm/max326xx/max32660-evsys/src/max326_button.c boards/arm/stm32/stm32f4discovery/src/stm32_gs2200m.c drivers/i2c/i2c_bitbang.c
drivers/power/supply/act8945a.c
drivers/serial/uart_pci_16550.c
drivers/serial/uart_pl011.c
drivers/wireless/bluetooth/bt_uart_shim.c

## Testing
ci

### CODE REVIEW

**Comment by nuttxpr** [comment]
[**\[Experimental Bot, please feedback here\]**](https://github.com/search?q=repo%3Aapache%2Fnuttx+13552&type=issues)



This PR appears to **partially meet** the NuttX requirements, but needs more information.

**Missing/Insufficient Information:**

* **Summary:**  "use small lock" is vague. *Why* are these changes necessary?  What problem does it solve? What kind of "small lock" is being used (e.g., a spinlock, mutex, semaphore)? Explain the rationale behind using this lock.  Provide a high-level overview of how the code is changed (e.g., "Replaced mutex with spinlock in interrupt handler for performance reasons"). Include relevant NuttX issue numbers if applicable.
* **Impact:** Just listing the files modified is not enough. Address *each* impact question specifically. For example:
    * **User Impact:**  Likely NO, but explain why.  Does this change behavior in a way the user might notice?
    * **Build Impact:** Likely NO, but confirm.  Are any new dependencies introduced?
    * **Hardware Impact:**  Specify affected architectures and boards.  Does this change driver behavior or add support for new hardware?
    * **Documentation Impact:**  Do any documentation updates need to be made to reflect these changes?
    * **Security Impact:**  Does using a different locking mechanism have security implications?  Justify your answer.
    * **Compatibility Impact:** Are there any backward compatibility concerns? Does this affect interoperability with other systems or software?
* **Testing:** "ci" is not sufficient. Provide details about your local setup (OS, compiler, etc.) and the target hardware used for testing. Include *actual* testing logs (even if brief) showing the behavior before and after the change.  Demonstrate that the changes work as intended and don't introduce regressions.  Ideally, the tests should exercise the code paths where the locking mechanisms are used.


**Example of Improved Impact Section:**

* **Impact:**
    * **User Impact:** NO.  This change is internal and should not affect user-visible behavior.
    * **Build Impact:** NO.  No changes to the build system are required.
    * **Hardware Impact:** YES. This change affects the following:
        * ARM: max326xx/max32660-evsys, stm32/stm32f4discovery
        * Xtensa: esp32s3
        * Z16: z16f
        * Z80: z8
        * PCI 16550 UART
        * PL011 UART
    * **Documentation Impact:** NO. No documentation updates are required for this internal change.
    * **Security Impact:**  Potentially YES.  Switching to a spinlock could lead to priority inversion if not handled carefully.  However, we have ensured that... [explain mitigation].
    * **Compatibility Impact:** NO.  This change maintains backward compatibility.


By providing more detail and addressing all the required points, the PR will be much easier to review and merge.

File: drivers/serial/uart_pl011.c [hunk 1]
```diff
@@ -248,6 +248,8 @@ static bool pl011_txempty(FAR struct uart_dev_s *dev);
  * Private Data
  ***************************************************************************/
 
+static spinlock_t g_uart_pl011_lock = SP_UNLOCKED;
```
Comment (xiaoxiang781216) [inline drivers/serial/uart_pl011.c:251]: move to pl011_uart_port_s

File: drivers/serial/uart_pci_16550.c [hunk 2]
```diff
@@ -121,6 +121,10 @@ static int pci_u16550_probe(FAR struct pci_device_s *dev);
  * Private Data
  *****************************************************************************/
 
+#ifdef CONFIG_16550_PCI_CONSOLE
+static spinlock_t g_uart_pci_lock = SP_UNLOCKED;
```
Comment (xiaoxiang781216) [inline drivers/serial/uart_pci_16550.c:125]: move to pci_u16550_priv_s

File: arch/z80/src/z8/z8_serial.c [hunk 3]
```diff
@@ -655,7 +664,15 @@ static void z8_txint(FAR struct uart_dev_s *dev, bool enable)
     }
 
   priv->txenabled = enable;
-  leave_critical_section(flags);
+}
+
+static void z8_txint(FAR struct uart_dev_s *dev, bool enable)
+{
+  struct z8_uart_s *priv    = (struct z8_uart_s *)dev->priv;
```
Comment (xiaoxiang781216) [inline arch/z80/src/z8/z8_serial.c:671]:
```suggestion
  struct z8_uart_s *priv = (struct z8_uart_s *)dev->priv;
```

File: arch/z80/src/z8/z8_serial.c [hunk 4]
```diff
@@ -638,10 +648,9 @@ static void z8_send(FAR struct uart_dev_s *dev, int ch)
  *
  ****************************************************************************/
 
-static void z8_txint(FAR struct uart_dev_s *dev, bool enable)
+static void z8_txint_nolock(FAR struct uart_dev_s *dev, bool enable)
 {
   struct z8_uart_s *priv    = (struct z8_uart_s *)dev->priv;
```
Comment (xiaoxiang781216) [inline arch/z80/src/z8/z8_serial.c:653]:
```suggestion
  struct z8_uart_s *priv = (struct z8_uart_s *)dev->priv;
```

File: arch/z80/src/z8/z8_serial.c [hunk 5]
```diff
@@ -599,7 +601,15 @@ static void z8_rxint(FAR struct uart_dev_s *dev, bool enable)
     }
 
   priv->rxenabled = enable;
-  leave_critical_section(flags);
+}
+
+static void z8_rxint(FAR struct uart_dev_s *dev, bool enable)
+{
+  struct z8_uart_s *priv    = (struct z8_uart_s *)dev->priv;
```
Comment (xiaoxiang781216) [inline arch/z80/src/z8/z8_serial.c:608]:
```suggestion
  struct z8_uart_s *priv = (struct z8_uart_s *)dev->priv;
```

File: arch/z80/src/z8/z8_serial.c [hunk 6]
```diff
@@ -582,10 +585,9 @@ static int z8_receive(FAR struct uart_dev_s *dev, FAR uint32_t *status)
  *
  ****************************************************************************/
 
-static void z8_rxint(FAR struct uart_dev_s *dev, bool enable)
+static void z8_rxint_nolock(FAR struct uart_dev_s *dev, bool enable)
 {
   struct z8_uart_s *priv    = (struct z8_uart_s *)dev->priv;
```
Comment (xiaoxiang781216) [inline arch/z80/src/z8/z8_serial.c:590]:
```suggestion
  struct z8_uart_s *priv = (struct z8_uart_s *)dev->priv;
```

File: arch/z80/src/z8/z8_serial.c [hunk 7]
```diff
@@ -255,16 +258,16 @@ static inline uint8_t z8_getuart(FAR struct z8_uart_s *priv, uint8_t offset)
 static uint8_t z8_disableuartirq(FAR struct uart_dev_s *dev)
 {
   struct z8_uart_s *priv    = (struct z8_uart_s *)dev->priv;
-  irqstate_t          flags = spin_lock_irqsave(NULL);
+  irqstate_t          flags = spin_lock_irqsave(&priv->lock);
```
Comment (xiaoxiang781216) [inline arch/z80/src/z8/z8_serial.c:261]:
```suggestion
  struct z8_uart_s *priv  = (struct z8_uart_s *)dev->priv;
  irqstate_t        flags = spin_lock_irqsave(&priv->lock);
  uint8_t           state = priv->rxenabled ?
                            STATE_RXENABLED : STATE_DISABLED | \
                            priv->txenabled ?
                            STATE_TXENABLED : STATE_DISABLED;
```

File: arch/z16/src/z16f/z16f_serial.c [hunk 8]
```diff
@@ -234,12 +237,13 @@ static uint8_t z16f_disableuartirq(struct uart_dev_s *dev)
 
 static void z16f_restoreuartirq(struct uart_dev_s *dev, uint8_t state)
 {
-  irqstate_t flags = spin_lock_irqsave(NULL);
+  struct z16f_uart_s *priv  = (struct z16f_uart_s *)dev->priv;
```
Comment (xiaoxiang781216) [inline arch/z16/src/z16f/z16f_serial.c:240]:
```suggestion
  struct z16f_uart_s *priv = (struct z16f_uart_s *)dev->priv;
```

File: arch/z16/src/z16f/z16f_serial.c [hunk 9]
```diff
@@ -507,10 +511,9 @@ static int z16f_receive(struct uart_dev_s *dev, uint32_t *status)
  *
  ****************************************************************************/
 
-static void z16f_rxint(struct uart_dev_s *dev, bool enable)
+static void z16f_rxint_nolock(struct uart_dev_s *dev, bool enable)
 {
   struct z16f_uart_s *priv  = (struct z16f_uart_s *)dev->priv;
```
Comment (xiaoxiang781216) [inline arch/z16/src/z16f/z16f_serial.c:516]:
```suggestion
  struct z16f_uart_s *priv = (struct z16f_uart_s *)dev->priv;
```

File: arch/z16/src/z16f/z16f_serial.c [hunk 10]
```diff
@@ -524,7 +527,15 @@ static void z16f_rxint(struct uart_dev_s *dev, bool enable)
     }
 
   priv->rxenabled = enable;
-  leave_critical_section(flags);
+}
+
+static void z16f_rxint(struct uart_dev_s *dev, bool enable)
+{
+  struct z16f_uart_s *priv  = (struct z16f_uart_s *)dev->priv;
```
Comment (xiaoxiang781216) [inline arch/z16/src/z16f/z16f_serial.c:534]:
```suggestion
  struct z16f_uart_s *priv = (struct z16f_uart_s *)dev->priv;
```

File: arch/z16/src/z16f/z16f_serial.c [hunk 11]
```diff
@@ -564,10 +575,9 @@ static void z16f_send(struct uart_dev_s *dev, int ch)
  *
  ****************************************************************************/
 
-static void z16f_txint(struct uart_dev_s *dev, bool enable)
+static void z16f_txint_nolock(struct uart_dev_s *dev, bool enable)
 {
   struct z16f_uart_s *priv  = (struct z16f_uart_s *)dev->priv;
```
Comment (xiaoxiang781216) [inline arch/z16/src/z16f/z16f_serial.c:580]:
```suggestion
  struct z16f_uart_s *priv = (struct z16f_uart_s *)dev->priv;
```

File: arch/z16/src/z16f/z16f_serial.c [hunk 12]
```diff
@@ -587,7 +597,15 @@ static void z16f_txint(struct uart_dev_s *dev, bool enable)
     }
 
   priv->txenabled = enable;
-  leave_critical_section(flags);
+}
+
+static void z16f_txint(struct uart_dev_s *dev, bool enable)
+{
+  struct z16f_uart_s *priv  = (struct z16f_uart_s *)dev->priv;
```
Comment (xiaoxiang781216) [inline arch/z16/src/z16f/z16f_serial.c:604]:
```suggestion
  struct z16f_uart_s *priv = (struct z16f_uart_s *)dev->priv;
```

Comment (hujun260) [reply arch/z16/src/z16f/z16f_serial.c]: done

**Review by xiaoxiang781216 — APPROVED** [review]

**Pull request merged**